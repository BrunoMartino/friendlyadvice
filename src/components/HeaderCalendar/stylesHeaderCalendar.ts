import styled, { css } from 'styled-components';
import {
  backgroundInpera10,
  backgroundLightSecundary,
  colorText,
} from '../../utils/colorsInpera';

export const Container = styled.div<any>`
  display: flex;
  flex: 1;
  justify-content: center;
  position: relative;

  padding: 0.8rem 2.4rem;

  ${({ calendarFlutter }) =>
    calendarFlutter &&
    css`
      padding: 0;
    `}

  align-items: center;
  color: ${colorText};
  background: transparent;

  .iconeCalendario {
    font-size: 3.2rem;
    cursor: pointer;
  }

  .MuiIconButton-root {
    nav-index: -1;
  }
`;

export const Calendar = styled.aside`
  display: flex;
  width: inherit;
  justify-content: center;

  > svg {
    margin: 0rem 0.8rem;
  }
`;

export const ContainerTwoCalendar = styled.div<any>`
  display: flex;
  width: 100%;
  justify-content: center;

  .iconeCalendar {
    font-size: 2.6rem;
    margin-right: 0.5rem;
    margin-top: 0.2rem;
  }

  .responsive-calendar {
    /* by setting font-size, all the elements will correspond */
    font-size: 9px !important; /* default to 10px */
  }

  @media (max-width: 1500px) {
    .responsive-calendar {
      font-size: 0.8rem !important;
    }
  }

  @media (max-width: 1200px) {
    .responsive-calendar {
      font-size: 0.7rem !important;
    }
  }

  @media (max-width: 768px) {
    .responsive-calendar {
      font-size: 0.6rem !important;
    }
  }

  /* Large screens */
  @media (min-width: 2500px) {
    .responsive-calendar {
      font-size: 1.2rem !important;
    }
  }

  .custom-today-day {
    background: none;
  }

  .my-custom-input-class {
    cursor: pointer;
    margin-right: 2.4rem;
    background: ${backgroundLightSecundary};
    border: 0.5px solid ${colorText};
    color: ${colorText};

    ${({ calendarFlutter }) =>
      calendarFlutter &&
      css`
        border: 2px solid #c0c0c0;
        color: #757575;
        background: white;
      `}

    outline: 'none';
  }

  .Calendar__day {
    color: ${colorText};
  }

  .Calendar__monthYear.-shown > *:hover,
  .Calendar:not(.-noFocusOutline) .Calendar__monthYear.-shown > *:focus,
  .Calendar__monthYear > *.-activeBackground {
    background: transparent;
  }

  .Calendar__day:not(.-blank):not(.-selectedStart):not(.-selectedEnd):not(.-selectedBetween):not(.-selected):hover {
    background: ${backgroundInpera10};
  }

  .calendarDatePicker {
    background: ${backgroundLightSecundary};
  }

  .DatePicker__input {
    width: 100%;
    height: 3.2rem;
    font-size: 1.4rem;
    font-weight: 600;

    ${({ calendarFlutter }) =>
      calendarFlutter &&
      css`
        height: 4rem;
        min-width: 20rem;
        font-weight: 400;
      `}
  }

  .Calendar__monthText
    .Calendar__yearText
    .Calendar__monthSelectorItemText
    .Calendar__yearSelectorText {
    color: ${colorText};
  }

  .Calendar__monthSelector,
  .Calendar__yearSelector {
    background-color: ${backgroundLightSecundary};
  }

  .Calendar__yearSelectorWrapper::after {
    background-image: none;
  }

  .Calendar__yearSelectorWrapper::before {
    background-image: none;
  }
`;
