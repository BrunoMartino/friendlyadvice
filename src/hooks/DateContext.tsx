import React, { createContext, useState, useContext } from 'react';
import { useSelector } from 'react-redux';
import { format } from 'date-fns';
import { dataLocal } from '../utils/fn';

interface IDataInicial {
  pDataInicial: Date | string;
}

interface IDataFinal {
  pDataFinal: Date | string;
}
interface IDataRange {
  pOBJDataRange: object;
}

interface DataContextData {
  dataInicial: Date | string | object;
  dataFinal: Date | string | object;

  atualizaData(credentials: IDataRange): Promise<string>;
}

export const DateContext = createContext<DataContextData>({
  dataInicial: new Date(),
  dataFinal: new Date(),
} as DataContextData);

export const DataContextoGeral: React.FC = ({ children }) => {
  const local = useSelector((state: any) => state.global.empresaAdmin.local);

  const [stDataInicial, setStDataInicial] = useState<IDataInicial>(() => {
    return {
      pDataInicial: format(dataLocal(local), 'yyyy-MM-dd'),
    };
  });

  const [stDataFinal, setStDataFinal] = useState<IDataFinal>(() => {
    return {
      pDataFinal: format(dataLocal(local), 'yyyy-MM-dd'),
    };
  });

  const atualizaData = async (pDataRange: any) => {
    if (pDataRange) {
      const { from, to } = pDataRange.pOBJDataRange;
      let dataInicial: string;
      let dataFinal: string;

      if (from) {
        dataInicial = format(
          new Date(from.year, from.month - 1, from.day),
          'yyyy-MM-dd',
        );
      } else {
        dataInicial = format(dataLocal(local), 'yyyy-MM-dd');
      }

      if (to) {
        dataFinal = format(
          new Date(to.year, to.month - 1, to.day),
          'yyyy-MM-dd',
        );
      } else {
        dataFinal = format(dataLocal(local), 'yyyy-MM-dd');
      }

      setStDataInicial({
        pDataInicial: dataInicial,
      });

      setStDataFinal({
        pDataFinal: dataFinal,
      });
    }

    return format(dataLocal(local), 'yyyy-MM-dd');
  };

  return (
    <DateContext.Provider
      value={{
        dataInicial: stDataInicial.pDataInicial,
        dataFinal: stDataFinal.pDataFinal,
        atualizaData,
      }}
    >
      {children}
    </DateContext.Provider>
  );
};

function useDate(): DataContextData {
  const context = useContext(DateContext);

  return context;
}

export { useDate };
