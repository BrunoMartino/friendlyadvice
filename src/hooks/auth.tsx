import React, { createContext, useCallback, useState, useContext } from 'react';
import jwt from 'jsonwebtoken';
import api from '../services/api';
import { useHistory } from 'react-router-dom';
import { useDispatch } from 'react-redux';
// import { abrirMensagem } from '../store/modules/SnackBar/action';
// import { TipoMensagem } from '../components/SnackBar/interface';
// import trataExcecao from '../utils/tratamentoExcecao';

interface Token {
  id: string;
}

interface AuthState {
  token: string;
}

interface SignInCredentials {
  email: string;
  senha: string;
}

interface AuthContextData {
  token: string | null;
  signIn(credentials: SignInCredentials): Promise<void>;
  signOut(): void;
}

const AuthContext = createContext<AuthContextData>({} as AuthContextData);

const AuthProvider: React.FC = ({ children }) => {
  // const dispatch = useDispatch();
  const history = useHistory();
  // DATA = Dados de Autenticação
  const [data, setData] = useState<AuthState>(() => {
    const token = localStorage.getItem('@INPERA:token');

    if (token) {
      try {
        const tokenCheck = jwt.verify(
          token,
          `${process.env.REACT_APP_JWT_PRIVATE_KEY}`,
        );

        if (tokenCheck) {
          api.defaults.headers.authorization = `Bearer ${token}`;
          return { token };
        }
      } catch (error) {
        return { token: `${error}` };
      }
    }

    return {} as AuthState;
  });

  // Função de login
  const signIn = useCallback(async ({ email, senha }) => {
    const response = await api.post('api/usuarios/login', {
      email,
      senha,
    });

    const { token } = response.data;
    const tokenCheck = jwt.verify(
      token,
      `${process.env.REACT_APP_JWT_PRIVATE_KEY}`,
    );

    if (tokenCheck) {
      localStorage.setItem('@INPERA:token', token);
      api.defaults.headers.authorization = `Bearer ${token}`;
      history.push('/listagem/demo');
    } else {
      throw new Error('Token não validado.');
    }
    setData({ token });
  }, []);

  // Função de fazer Logout
  const signOut = useCallback(() => {
    localStorage.removeItem('@INPERA:token');
    history.replace('/');

    setData({} as AuthState);
  }, [history]);

  return (
    <AuthContext.Provider value={{ token: data.token, signIn, signOut }}>
      {children}
    </AuthContext.Provider>
  );
};

function useAuth(): AuthContextData {
  const context = useContext(AuthContext);

  return context;
}

export { AuthProvider, useAuth };
