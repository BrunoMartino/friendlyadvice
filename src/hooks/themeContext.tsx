import React, { createContext, useCallback, useState, useContext } from 'react';

interface ThemeContextData {
  saveTheme(theme: string): void;
  theme: string;
}

interface ThemeState {
  theme: string;
}

const ThemeContext = createContext<ThemeContextData>({} as ThemeContextData);

export const ThemeProvider: React.FC = ({ children }) => {
  const [data, setData] = useState<ThemeState>(() => {
    const theme = localStorage.getItem('theme');

    const tema = JSON.parse(theme as string);

    if (tema) {
      return tema.title;
    }

    return {} as ThemeState;
  });

  const saveTheme = useCallback((theme) => {
    setData({ theme });
  }, []);

  return (
    <ThemeContext.Provider
      value={{
        saveTheme,
        theme: data.theme,
      }}
    >
      {children}
    </ThemeContext.Provider>
  );
};

export function useTheme(): ThemeContextData {
  const context = useContext(ThemeContext);

  if (!context) {
    throw new Error('useTheme must be used within an AuthProvider');
  }

  return context;
}
