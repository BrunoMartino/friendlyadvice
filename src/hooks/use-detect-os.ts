export const useDetectOS = () => {
  var isMobile = {
    Android: function () {
      return navigator.userAgent.match(/Android/i);
    },
    iOS: function () {
      return navigator.userAgent.match(/iPhone|iPad|iPod/i);
    },
    any: function () {
      return isMobile.Android() || isMobile.iOS();
    },
  };

  let whichOS = '';

  if (!isMobile.any()) {
    //Trigger only in the case of NO Mobile operating system detected
    whichOS = 'not detected';
  }
  if (isMobile.any()) {
    //Trigger on any Mobile Operating System.
    whichOS = 'any mobile detected';
  }
  if (isMobile.Android()) {
    //Trigger only on Android Devices
    whichOS = 'android';
  }
  if (isMobile.iOS()) {
    //Trigger only on iOS Devices
    whichOS = 'ios';
  }

  return whichOS !== '' && whichOS;
};
