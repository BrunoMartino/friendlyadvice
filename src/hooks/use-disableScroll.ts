import { useEffect } from 'react';

const keys: any = { 37: 1, 38: 1, 39: 1, 40: 1 };

function preventDefault(e: Event) {
  e.preventDefault();
}

function preventDefaultForScrollKeys(e: any) {
  if (keys[e.keyCode]) {
    preventDefault(e);
    return false;
  }
}

export const useDisableScroll = () => {
  useEffect(() => {
    disableScroll();
    return () => enableScroll();
  }, []);

  const disableScroll = () => {
    document.body.style.overflow = 'hidden';
    window.addEventListener('keydown', preventDefaultForScrollKeys, false);
  };

  const enableScroll = () => {
    document.body.style.overflow = '';
    window.removeEventListener('keydown', preventDefaultForScrollKeys, false);
  };

  return {
    disableScroll,
    enableScroll,
  };
};
