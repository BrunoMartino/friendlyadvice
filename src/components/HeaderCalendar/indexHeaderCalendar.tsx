import React, { useCallback, useContext, useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { registerLocale } from 'react-datepicker';
import ptBR from 'date-fns/locale/pt-BR';
import DatePicker, { DayRange } from 'react-modern-calendar-datepicker';
import { addMonths, getDate, getMonth, getYear } from 'date-fns';
import moment from 'moment';
import { FaRegCalendarAlt } from 'react-icons/fa';
import { formatarData, myCustomLocale } from '../../utils/fn';
import 'react-modern-calendar-datepicker/lib/DatePicker.css';
import {
  Container,
  Calendar,
  ContainerTwoCalendar,
} from './stylesHeaderCalendar';
import 'react-day-picker/lib/style.css';
import { DateContext } from '../../hooks/DateContext';
import { abrirMensagem } from '../../store/modules/Components/SnackBar/action';
import { TipoMensagem } from '../SnackBar/interface';
import { backgroundInpera, backgroundInpera10 } from '../../utils/colorsInpera';
import MenuSlide from '../Menus/MenuSlide/MenuSlide.Index';

interface IMinMaxDate {
  year: number;
  month: number;
  day: number;
}

const HeaderCalendar = ({
  hideIcon = false,
  calendarFlutter = false,
  resetToToday,
  emptyCalendar = false,
  limitRange = false,
  selectedDate,
}: {
  hideIcon?: boolean;
  calendarFlutter?: boolean;
  resetToToday?: boolean;
  emptyCalendar?: boolean;
  limitRange?: boolean;
  selectedDate?: DayRange;
}) => {
  const dispatch = useDispatch();
  const setLoading = useSelector(
    (state: any) => state.global.empresaAdmin.loading,
  );

  const [minDate, setMinDate] = useState<IMinMaxDate>({
    year: 0,
    month: 0,
    day: 0,
  });

  const [maxDate, setMaxDate] = useState<IMinMaxDate>({
    year: 0,
    month: 0,
    day: 0,
  });

  const { atualizaData } = useContext(DateContext);
  /* istanbul ignore next */
  const handlerData = (dat: any) => {
    if (!dat.from) return;

    const formatarDataRange = (data: {
      day: number;
      month: number;
      year: number;
    }) =>
      formatarData({
        dia: data.day,
        mes: data.month - 1,
        ano: data.year,
        formato: 'yyyy-MM-dd',
      });

    if (limitRange) {
      const dataInicialCalculo = formatarDataRange(dat.from);

      if (dat.to) {
        setMinDate({ year: 1900, month: 0, day: 1 });
        setMaxDate({ year: 2100, month: 11, day: 31 });
      } else {
        const dataRef = new Date(dataInicialCalculo);
        const vMinDate = addMonths(dataRef, -2);
        const vMaxDate = addMonths(dataRef, 2);

        setMinDate({
          year: getYear(vMinDate),
          month: getMonth(vMinDate) + 1,
          day: getDate(vMinDate),
        });

        setMaxDate({
          year: getYear(vMaxDate),
          month: getMonth(vMaxDate) + 1,
          day: getDate(vMaxDate),
        });
      }

      if (dat.to) {
        const dataFinalCalculo = formatarDataRange(dat.to);
        const compareDays = moment(dataFinalCalculo).diff(
          dataInicialCalculo,
          'days',
        );

        if (compareDays <= 62) {
          atualizaData({ pOBJDataRange: dat });
        } else {
          dispatch(
            abrirMensagem({
              open: true,
              mensagem: 'Data superior a 60 dias!',
              tipo: TipoMensagem.ERRO,
            }),
          );
        }
      }
    } else {
      if (dat.from || dat.to) {
        atualizaData({ pOBJDataRange: dat });
      } else {
        dispatch(
          abrirMensagem({
            open: true,
            mensagem: 'Data inválida!',
            tipo: TipoMensagem.ERRO,
          }),
        );
      }
    }
  };
  // eslint-disable-next-line react-hooks/exhaustive-deps

  /* istanbul ignore next */
  const [selectedDayRange, setSelectedDayRange] = useState<DayRange>({
    from: emptyCalendar
      ? null
      : {
          day: getDate(new Date()),
          month: getMonth(new Date()) + 1,
          year: getYear(new Date()),
        },
    to: emptyCalendar
      ? null
      : {
          day: getDate(new Date()),
          month: getMonth(new Date()) + 1,
          year: getYear(new Date()),
        },
  });

  const formatInputValue = () => {
    if (selectedDayRange.from && selectedDayRange.to) {
      return ` ${formatarData({
        dia: selectedDayRange.from?.day!,
        mes: selectedDayRange.from?.month! - 1,
        ano: selectedDayRange.from?.year!,
        formato: 'dd/MM/yyyy',
      })} ➥ ${formatarData({
        dia: selectedDayRange.to?.day!,
        mes: selectedDayRange.to?.month! - 1,
        ano: selectedDayRange.to?.year!,
        formato: 'dd/MM/yyyy',
      })}`;
    }
    if (selectedDayRange.from) {
      return ` ${formatarData({
        dia: selectedDayRange.from?.day!,
        mes: selectedDayRange.from?.month! - 1,
        ano: selectedDayRange.from?.year!,
        formato: 'dd/MM/yyyy',
      })}`;
    }

    return '';
  };

  useEffect(() => {
    setSelectedDayRange({
      from:
        selectedDate && selectedDate.from !== null
          ? {
              day: Number(selectedDate.from!.day),
              month: Number(selectedDate.from!.month),
              year: Number(selectedDate.from!.year),
            }
          : emptyCalendar
          ? null
          : {
              day: getDate(new Date()),
              month: getMonth(new Date()) + 1,
              year: getYear(new Date()),
            },
      to:
        selectedDate && selectedDate.to !== null
          ? {
              day: Number(selectedDate.to!.day),
              month: Number(selectedDate.to!.month),
              year: Number(selectedDate.to!.year),
            }
          : emptyCalendar
          ? null
          : {
              day: getDate(new Date()),
              month: getMonth(new Date()) + 1,
              year: getYear(new Date()),
            },
    });
  }, [resetToToday]);

  useEffect(() => {
    handlerData(selectedDayRange);
  }, [selectedDayRange]);

  registerLocale('ptBR', ptBR);

  return (
    <Container calendarFlutter={calendarFlutter}>
      {!hideIcon && !setLoading && (
        <div
          style={{
            marginRight: '1rem',
            paddingBottom: '4rem',
            height: '100%',
          }}
        >
          <MenuSlide />
        </div>
      )}
      <Calendar data-testid="Calendar">
        <ContainerTwoCalendar calendarFlutter={calendarFlutter}>
          {/* <FaRegCalendarAlt className="iconeCalendar" /> */}
          {!hideIcon ? <FaRegCalendarAlt className="iconeCalendar" /> : null}
          <DatePicker
            value={selectedDayRange}
            onChange={setSelectedDayRange}
            inputPlaceholder="Selecionar período"
            shouldHighlightWeekends={false}
            locale={myCustomLocale}
            inputClassName="my-custom-input-class"
            calendarClassName="calendarDatePicker"
            formatInputText={formatInputValue}
            minimumDate={limitRange ? minDate : undefined}
            maximumDate={limitRange ? maxDate : undefined}
            calendarTodayClassName="custom-today-day"
            calendarPopperPosition="bottom"
            colorPrimary={backgroundInpera}
            colorPrimaryLight={backgroundInpera10}
            wrapperClassName="datePicker"
          />
        </ContainerTwoCalendar>
      </Calendar>
    </Container>
  );
};

export default HeaderCalendar;
