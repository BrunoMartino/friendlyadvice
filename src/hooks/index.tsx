import React from 'react';

import {ModalProvider} from 'styled-react-modal';
import {AuthProvider} from './auth';
import {ThemeProvider} from './themeContext';
import {DataContextoGeral} from './DateContext';

const AppProvider: React.FC = ({children}) => (
  <AuthProvider>
    <DataContextoGeral>
      <ThemeProvider>
        <ModalProvider>{children}</ModalProvider>
      </ThemeProvider>
    </DataContextoGeral>
  </AuthProvider>
);

export default AppProvider;
