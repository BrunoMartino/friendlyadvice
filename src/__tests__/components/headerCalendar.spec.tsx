import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import { act } from 'react-test-renderer';

import { renderHook } from '@testing-library/react-hooks';
import HeaderCalendar from '../../components/HeaderCalendar/indexHeaderCalendar';
import { DataContextoGeral, useDate } from '../../hooks/DateContext';

const spyGetItem = jest.spyOn(Storage.prototype, 'getItem');

spyGetItem.mockImplementation((key) => {
  switch (key) {
    case '@INPERA:token':
      return '1234';

    case 'theme':
      return JSON.stringify({ title: 'dark' });

    default:
      return null;
  }
});

describe('headerCalendar Component', () => {
  it('should be able to render header Calendar', () => {
    const { debug } = render(
      <DataContextoGeral>
        <HeaderCalendar />
      </DataContextoGeral>,
    );

    debug();
  });

  it('should be able new Date()', () => {
    const { result } = renderHook(() => useDate(), {
      wrapper: DataContextoGeral,
    });
  });

  it('should be able change Date', () => {
    const { result } = renderHook(() => useDate(), {
      wrapper: DataContextoGeral,
    });

    act(() => {
      result.current.atualizaData({
        pOBJDataRange: {},
      });
    });

    const dt = new Date();

    const dtDateOnly = new Date(
      dt.valueOf() + dt.getTimezoneOffset() * 60 * 1000,
    );
  });
});
