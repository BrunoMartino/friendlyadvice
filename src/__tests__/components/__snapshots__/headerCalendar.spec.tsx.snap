// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`headerCalendar Component should match DateRangeFilter snapshot 1`] = `
<div
  className="sc-gtssRu iQqsPf"
>
  <div
    className="sc-dlnjPT hZRtBM"
  >
    <div>
      Dashboard
    </div>
  </div>
  <aside
    className="sc-hKFyIo eglASt"
    data-testid="Calendar"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
    >
      <div
        className="sc-eCApGN jZwxkW"
      >
        <div>
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
            data-testid="dataInicial"
          >
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
                disabled={false}
                id="date-picker-Inicial"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="dd/mm/aaaa"
                required={false}
                type="tel"
                value="24/09/2020"
              />
              <div
                className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
              >
                <button
                  aria-label="change date"
                  className="MuiButtonBase-root MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
            
          </div>
        </div>
        <div
          className="separadorCalendarios"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
            data-testid="dataFinal"
          >
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
                disabled={false}
                id="date-picker-Final"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="dd/mm/aaaa"
                required={false}
                type="tel"
                value="24/09/2020"
              />
              <div
                className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
              >
                <button
                  aria-label="change date"
                  className="MuiButtonBase-root MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
            
          </div>
        </div>
      </div>
    </div>
  </aside>
</div>
`;
